### ISSUE TITLE
Support "-opt-in": Build results in "warning: '-Xuse-experimental' is deprecated..."

### ISSUE THREAD

**Issue by brettwooldridge** [bazelbuild/rules_kotlin#687]
I am converting a project from Buck to Bazel, and it is mostly going smoothly, but build is emitting this warning:
```
INFO: From KotlinCompile //netld/org.ziptie.net:com-dancer-net { kt: 4, java: 30, srcjars: 0 } for darwin:
warning: '-Xuse-experimental' is deprecated and will be removed in a future release
```
My toolchain config looks like this:
```
load("@io_bazel_rules_kotlin//kotlin:core.bzl", "define_kt_toolchain")

define_kt_toolchain(
    name = "kotlin_toolchain",
    api_version = "1.5",
    jvm_target = "11",
    language_version = "1.5",
)
```
Running ``kotlin -version`` on the CLI returns:
```
Kotlin version 1.5.31-release-548 (JRE 17.0.2+8-LTS-86)
```
How can I get rid of the ``'-Xuse-experimental'`` warning?

**Comment by brettwooldridge** [comment]
Replying to myself. This seems to eliminate the warning:
```
kt_kotlinc_options(
    name = "kt_kotlinc_options",
    x_use_experimental = False,
)

define_kt_toolchain(
    name = "kotlin_toolchain",
    api_version = "1.4",
    jvm_target = "11",
    kotlinc_options = "//:kt_kotlinc_options",
    language_version = "1.4",
)
```
However, this may turn off needed features. Kotlin is apparently now using ``-opt-in`` from 1.6 and ``-Xopt-in`` in earlier versions.

https://kotlinlang.org/docs/opt-in-requirements.html

**Comment by Kernald** [comment]
See #676.
